Lib/ldap/dn.py

```diff
@@ -48,18 +48,19 @@ def str2dn(dn,flags=0):
   return ldap.functions._ldap_function_call(None,_ldap.str2dn,dn,flags)
 
 
-def dn2str(dn):
+def dn2str(dn, flags=_ldap.DN_FORMAT_LDAPV3):
   """
   This function takes a decomposed DN as parameter and returns
-  a single string. It's the inverse to str2dn() but will always
-  return a DN in LDAPv3 format compliant to RFC 4514.
+  a single string. It's the inverse to str2dn() but will by default always
+  return a DN in LDAPv3 format compliant to RFC 4514 if not otherwise specified
+  via flags.
+
+  See also the OpenLDAP man-page ldap_dn2str(3)
   """
-  return ','.join([
-    '+'.join([
-      '='.join((atype,escape_dn_chars(avalue or '')))
-      for atype,avalue,dummy in rdn])
-    for rdn in dn
-  ])
+  #if not dn:
+  #    return ''
```

Comment:
required?

Reply:
libldap isn't too sure either, but allows it: https://git.openldap.org/openldap/openldap/-/blob/OPENLDAP_REL_ENG_2_6_3/libraries/libldap/getdn.c#L3021-3028

Reply:
Unless we *want* to make the distinction for `None` and make `dn2str(None) == None`, might be too much hassle though.